.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "libs/azure-iot-middleware-freertos"]
+	path = libs/azure-iot-middleware-freertos
+	url = git@github.com:Azure/azure-iot-middleware-freertos.git
```

Comment:
I don't know if I've ever commented on a `.gitmodules` file so this is a first for me :)

Can we make this the http version of the repo for greater compatibility on pulling it? It could potentially complicate the process for contributing as someone would have to set up their username/password by default but it increases the compatibility if they don't have ssh set up. Plus they can always change the url to the ssh one if they want to contribute and use ssh.